modules/github_integration/lib/api/v3/github_integration/github_pull_requests_by_work_package_api.rb

```diff
@@ -0,0 +1,52 @@
+#-- encoding: UTF-8
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+module API
+  module V3
+    module GithubIntegration
+      class GithubPullRequestsByWorkPackageAPI < ::API::OpenProjectAPI
+        after_validation do
+          authorize(:show_github_content, context: @work_package.project)
+          @github_pull_requests = @work_package.github_pull_requests
+        end
+
+        resources :github_pull_requests do
+          get do
+            path = api_v3_paths.github_pull_requests_by_work_package(@work_package.id)
+            GithubPullRequestCollectionRepresenter.new(@github_pull_requests,
+                                                       @github_pull_requests.count,
+                                                       self_link: path,
+                                                       current_user: current_user)
+          end
```

Comment:
We try to heavily standardize to controllers to keep them slim. Because of this, we have created in 'Index` endpoint class. See https://github.com/opf/openproject/blob/dev/modules/costs/lib/api/v3/time_entries/available_projects_api.rb for an example.
It deduces a lot so if the expected schema is not matched, it might be hard to get it to run at first. But having a strictly expected schema is actually one of the main intends of doing it this way.

Reply:
Thanks for this hint. I attempted to use the index endpoint class (and am really not sure if my code is more slim now with all the boilerplate classes I needed to add 🤷 ). Anyways, there is an error I need your advise on

<img width="1023" alt="Screenshot 2021-04-12 at 23 33 41" src="https://user-images.githubusercontent.com/206108/114465565-8ff6d100-9be7-11eb-988c-5c3efeabebe2.png">

It seems to want a static path from me, something like `api/v3/github_pull_requests`. But I can't provide this, due to permissions (see above). do you have an idea?

Reply:
I mean I could (for every `GithubPullRequest`) find the list of `work_packages` and for each `WorkPackage` it's `project`. And if any of these projects grants the `current_user` the `show_github_content` permission, I could show the PR. But that sounds very inefficient 🤔

Reply:
It seems it is not yet possible to overwrite the self_path for the index as it would need to be done here. So don't make that change for now. I'll do it later on when I find the time.

Reply:
alright :) thanks for investigating 💚